packages/core/admin/admin/src/content-manager/components/RelationInputDataManager/RelationInputDataManager.js

```diff
@@ -34,14 +38,26 @@ export const RelationInputDataManager = ({
   const { connectRelation, disconnectRelation, loadRelation, modifiedData, slug, initialData } =
     useCMEditViewDataManager();
 
+  const relationsFromModifiedData = get(modifiedData, name) ?? [];
+
+  const currentLastPage = Math.ceil(relationsFromModifiedData.length / RELATIONS_TO_DISPLAY);
+
   const { relations, search, searchFor } = useRelation(`${slug}-${name}-${initialData?.id ?? ''}`, {
+    name,
     relation: {
       enabled: get(initialData, name)?.count !== 0 && !!endpoints.relation,
       endpoint: endpoints.relation,
+      pageGoal: currentLastPage,
       pageParams: {
         ...defaultParams,
         pageSize: RELATIONS_TO_DISPLAY,
       },
+      onLoadRelationsCallback: loadRelation,
```

Comment:
I'd suggest to change the method name, as it is already scoped to relations.

```suggestion
      onLoad: loadRelation,
```